BaragonAgentService/src/main/java/com/hubspot/baragon/agent/managers/AgentRequestManager.java

```diff
@@ -225,6 +241,51 @@ public Response getRenderedConfigs(String serviceId) {
     return Response.ok(result).build();
   }
 
+
+  public static String getServiceBasePathWithoutLeadingSlash(String path){
+    if (path.startsWith("/")){
+      path = path.replaceFirst("/", "");
+    }
+    return path;
+  }
+
+  public Response purgeCache(String serviceId) {
+    LOG.info("purgeCache() called with serviceId={}", serviceId);
+
+    Optional<BaragonService> maybeService;
+    if (internalStateCache.containsKey(serviceId)){
+      maybeService = Optional.of(internalStateCache.get(serviceId).getService());
+    }
+    else {
+      maybeService = stateDatastore.getService(serviceId);
+    }
+
+    // 1. if no service exists, return an error response early
+    if (!maybeService.isPresent()){
+      return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(String.format("Could not find service with serviceId=%s", serviceId)).build();
+    }
+
+    // 2. remove a leading slash if it exists and generate the URI
+    String serviceBasePathWithoutLeadingSlash = getServiceBasePathWithoutLeadingSlash(
```

Comment:
Do we want to purge the cache by path or by service id or maybe by both?
- When we deploy and want to purge the cache, that is generally for a particular service and all routes for it
- For some of our services, the routes can be things like regexes, which ends up tricky to deal with
- Does the cache in nginx cache by individual route and tag by service? i.e. can the module we are adding handle 'purge all things tagged with serviceId'?

Reply:
Hmmm, this has me thinking, the module supports wildcard requests at the end, so we could do something like set a `service_id` variable and then set `proxy_cache_key $service_id$uri$is_args$args;` and then when a purge request comes in just call purge on `$service_id*` which would clear the whole service.

Reply:
Yeah, that'd be great. Maybe we have some sort of divider in there as well between service_id and the rest as well (thinking of a case like MyThingUI and MyThingWithMoreStuffUI as service ids, first would also purge the second)

Reply:
Would have to be careful about a service_id containing another service_id. Maybe we generate a hash from Baragon's service_id and use that instead of the raw service_id